tezos/context/src/tezedge_context.rs

```diff
@@ -746,7 +836,7 @@ impl IndexApi<TezedgeContext> for TezedgeIndex {
 pub struct TezedgeContext {
     /// Index used for fetching and saving objects from/to the repository.
     pub index: TezedgeIndex,
-    pub parent_commit_hash: Option<HashId>,
+    pub parent_commit_ref: Option<ObjectReference>,
```

Comment:
Maybe it is ok this way, but in my mind the reference would be just the hash (we have an index that maps hashes to commit objects).